app/components/startContainer/startContainerController.js

```
@@ -1,41 +1,105 @@
+        config.HostConfig.DnsSearch = getNames(config.HostConfig.DnsSearch);
+        config.HostConfig.CapAdd = getNames(config.HostConfig.CapAdd);
+        config.HostConfig.CapDrop = getNames(config.HostConfig.CapDrop);
+
+        var ExposedPorts = {};
+        var PortBindings = {};
+        // TODO: consider using compatibility library 
+        config.HostConfig.PortBindings.forEach(function(portBinding) {
+            var intPort = portBinding.intPort + "/tcp";
+            var binding = {
+                HostIp: portBinding.ip,
+                HostPort: portBinding.extPort
+            };
+            if (portBinding.intPort) {
+                ExposedPorts[intPort] = {};
+                if (intPort in PortBindings) {
+                    PortBindings[intPort].push(binding);
+                } else {
+                    PortBindings[intPort] = [binding];
+                }
+            } else {
+                // TODO: Send warning message? Internal port need to be specified.
+            }
+        });
+        config.ExposedPorts = ExposedPorts;
+        config.HostConfig.PortBindings = PortBindings;
+
+        // Remove empty fields from the request to avoid overriding defaults
+        rmEmptyKeys(config.HostConfig);
+        rmEmptyKeys(config);
+
         var ctor = Container;
         var loc = $location;
         var s = $scope;
-
-        Container.create({
-                Image: id,
-                name: $scope.config.name,
-                Memory: $scope.config.memory,
-                MemorySwap: $scope.config.memorySwap,
-                CpuShares: $scope.config.cpuShares,
-                Cmd: cmds,
-                VolumesFrom: $scope.config.volumesFrom
-            }, function(d) {
+        Container.create(config, function(d) {
                 if (d.Id) {
                     ctor.start({id: d.Id}, function(cd) {
                         $('#create-modal').modal('hide');
```

TOOD: Start errors are ignored here.